tools/utils/cloud-image-downloader.sh

```diff
@@ -0,0 +1,252 @@
+    ["ubuntu-22.04-server-cloudimg-arm64.img"]="https://cloud-images.ubuntu.com/releases/22.04/release/ubuntu-22.04-server-cloudimg-arm64.img"
+    ["ubuntu-20.04-server-cloudimg-arm64.img"]="https://cloud-images.ubuntu.com/releases/20.04/release/ubuntu-20.04-server-cloudimg-arm64.img"
+    ["OL9U5_x86_64-kvm-b259.qcow2"]="https://yum.oracle.com/templates/OracleLinux/OL9/u5/x86_64/OL9U5_x86_64-kvm-b259.qcow2"
+    ["OL8U10_x86_64-kvm-b258.qcow2"]="https://yum.oracle.com/templates/OracleLinux/OL8/u10/x86_64/OL8U10_x86_64-kvm-b258.qcow2"
+    ["OL9U5_aarch64-kvm-b126.qcow2"]="https://yum.oracle.com/templates/OracleLinux/OL9/u5/aarch64/OL9U5_aarch64-kvm-b126.qcow2"
+    ["OL8U10_aarch64-kvm-b122.qcow2"]="https://yum.oracle.com/templates/OracleLinux/OL8/u10/aarch64/OL8U10_aarch64-kvm-b122.qcow2"
+    ["Rocky-8-GenericCloud.latest.x86_64.qcow2"]="https://dl.rockylinux.org/pub/rocky/8/images/x86_64/Rocky-8-GenericCloud.latest.x86_64.qcow2"
+)
+
+# 2. Add the destination filename and its corresponding distribution subdirectory name.
+declare -A IMAGE_DISTROS=(
+    ["Rocky-8-GenericCloud.latest.aarch64.qcow2"]="rockylinux"
+    ["Rocky-8-GenericCloud.latest.x86_64.qcow2"]="rockylinux"
+    ["Rocky-9-GenericCloud.latest.aarch64.qcow2"]="rockylinux"
+    ["Rocky-9-GenericCloud.latest.x86_64.qcow2"]="rockylinux"
+    ["openSUSE-Leap-15.5-Minimal-VM.x86_64-Cloud.qcow2"]="opensuse"
+    ["openSUSE-Leap-15.5-Minimal-VM.aarch64-Cloud.qcow2"]="opensuse"
+    ["debian-12-genericcloud-amd64.qcow2"]="debian"
+    ["debian-12-genericcloud-arm64.qcow2"]="debian"
+    ["ubuntu-24.04-server-cloudimg-amd64.img"]="ubuntu"
+    ["ubuntu-22.04-server-cloudimg-amd64.img"]="ubuntu"
+    ["ubuntu-20.04-server-cloudimg-amd64.img"]="ubuntu"
+    ["ubuntu-24.04-server-cloudimg-arm64.img"]="ubuntu"
+    ["ubuntu-22.04-server-cloudimg-arm64.img"]="ubuntu"
+    ["ubuntu-20.04-server-cloudimg-arm64.img"]="ubuntu"
+    ["OL9U5_x86_64-kvm-b259.qcow2"]="oraclelinux"
+    ["OL8U10_x86_64-kvm-b258.qcow2"]="oraclelinux"
+    ["OL9U5_aarch64-kvm-b126.qcow2"]="oraclelinux"
+    ["OL8U10_aarch64-kvm-b122.qcow2"]="oraclelinux"
+)
+
+#-------------------------------------------------------------------------------
+# Cleanup Handler
+#-------------------------------------------------------------------------------
+
+cleanup_on_exit() {
+    local exit_code=$?
+    if [ -d "$TEMP_DIR" ]; then
+        rm -rf "$TEMP_DIR"
+        log_info "Temporary directory $TEMP_DIR removed."
+    fi
+
+    if [ $exit_code -ne 0 ]; then
+        log_error "Script exited with error code: $exit_code"
+    fi
+}
+
+trap cleanup_on_exit EXIT INT TER
+
+#-------------------------------------------------------------------------------
+# Main Script Logic
+#-------------------------------------------------------------------------------
+
+if ! command -v logger &> /dev/null; then
+    log_warn "logger utility not found - syslog logging disabled"
+fi
```

Comment:
shouldn’t a flag be set and the call to `logger` in `log_message()` be made conditional?